lib/install.js

```diff
@@ -95,69 +116,120 @@ install.completion = function (opts, cb) {
   cb()
 }
 
-const Arborist = require('@npmcli/arborist')
-
-// dependencies
-var log = require('npmlog')
-// const sillyLogTree = require('./util/silly-log-tree.js')
-
-// npm internal utils
-var npm = require('./npm.js')
-var output = require('./utils/output.js')
-var saveMetrics = require('./utils/metrics.js').save
-
-// install specific libraries
-var audit = require('./install/audit.js')
-var {
-  getPrintFundingReport,
-  getPrintFundingReportJSON
-} = require('./install/fund.js')
-var errorMessage = require('./utils/error-message.js')
-
-const path = require('path')
-
-function install (where, args, cb) {
+async function install (where, args, cb) {
   if (!cb) {
     cb = args
     args = where
     where = null
   }
+
   // the /path/to/node_modules/..
   const globalTop = path.resolve(npm.globalDir, '..')
-  if (!where) {
-    where = npm.flatOptions.global
+  const {
+    dryRun,
+    global: isGlobalInstall
+  } = npm.flatOptions
+
+  // Determine where install is happening
+  // - If `where` is NOT set, then we need to determine if it's a global install
+  //    or if it's a local context install (hence nested ternary)
+  // - `where` IS SET, just use it
+  where = (!where)
+    ? (isGlobalInstall)
       ? globalTop
       : npm.prefix
-  }
-  const {dryRun} = npm.flatOptions
+    : where
 
   // TODO: Add warnings for other deprecated flags
   if (npm.config.get('dev')) {
     log.warn('install', 'Usage of the `--dev` option is deprecated. Use `--include=dev` instead.')
   }
 
+  // If global install, and NO args passed; user meant "this directory context"
   if (where === globalTop && !args.length) {
     args = ['.']
   }
+
+  // XXX What does this do?
   args = args.filter(a => path.resolve(a) !== npm.prefix)
 
-  const arb = new Arborist({
-    ...this.flatOptions,
-    path: where,
-  })
+  // Create arborist object to work with
+  const arb = new Arborist({ ...npm.flatOptions, path: where })
 
   // TODO:
   // - audit
   // - funding
-  // - more logging (archy-ize the tree for silly logging)
   // - global installs in Arborist
 
-  const opt = {
-    ...this.flatOptions,
-    add: args,
-  }
-  arb[dryRun ? 'buildIdealTree' : 'reify'](opt).then(tree => {
-    output('TREEEEEEEE', tree)
+  const arbCallOpts = { ...npm.flatOptions, add: args }
+
+  try {
+    const start = Date.now()
```

Comment:
I'd love opinions on this `start`/`stop` logic. I don't love it, but I was trying to keep the output from the previous version of the command.

Reply:
Yeah, Date.now() is a bit blunt, but probably fine enough accuracy for our purposes.  It'd of course be more precise to use `process.hrtime()`, and get nanosecond precision, but since big installs are sometimes several minutes long, we probably don't need that much precision.